scripts/cast-on-tenderly/index.js

```diff
@@ -0,0 +1,87 @@
+import 'dotenv/config';
+import axios from 'axios';
+import { Contract, ethers } from 'ethers';
+
+const NETWORK_ID = 5;
+const CHIEF_ADDRESS = '0x33Ed584fc655b08b2bca45E1C5b5f07c98053bC1';
+const CHIEF_HAT_SLOT = 12;
+const DEFAULT_TRANSACTION_PARAMETERS = { gasLimit: 1000000000 };
+
+// check env vars
+const REQUIRED_ENV_VARS = ['TENDERLY_USER', 'TENDERLY_PROJECT', 'TENDERLY_ACCESS_KEY'];
+if (REQUIRED_ENV_VARS.some(varName => !process.env[varName])) {
+    throw new Error(`Please provide all required env variables: ${REQUIRED_ENV_VARS.join(', ')}`);
+}
+
+// check process arguments
+const SPELL_ADDRESS = process.argv[2];
+if (!SPELL_ADDRESS) {
+    throw new Error('Please provide address of the spell, e.g.: `node index.js 0x...`');
+}
+
+const makeTenderlyApiRequest = async function (path) {
+    const API_BASE = `https://api.tenderly.co/api/v1/account/${process.env.TENDERLY_USER}/project/${process.env.TENDERLY_PROJECT}`;
+    return await axios.post(
+        `${API_BASE}${path}`,
+        { network_id: NETWORK_ID },
+        { headers: { 'X-Access-Key': process.env.TENDERLY_ACCESS_KEY } }
+    );
+};
+
+const createTenderlyFork = async function () {
+    const response = await makeTenderlyApiRequest('/fork');
+    const forkId = response.data.simulation_fork.id;
+    const rpcUrl = `https://rpc.tenderly.co/fork/${forkId}`;
+    const forkUrl = `https://dashboard.tenderly.co/${process.env.TENDERLY_USER}/${process.env.TENDERLY_PROJECT}/fork/${forkId}`;
+    return { forkId, forkUrl, rpcUrl };
+};
+
+const publishTenderlyTransaction = async function (forkId, transactionId) {
+    await makeTenderlyApiRequest(`/fork/${forkId}/transaction/${transactionId}/share`);
+    return `https://dashboard.tenderly.co/shared/fork/simulation/${transactionId}`;
+};
+
+const runSpell = async function () {
+    const { forkId, forkUrl, rpcUrl } = await createTenderlyFork();
+    console.info('private tenderly fork is created', forkUrl);
+
+    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    const signer = provider.getSigner();
+
+    console.info('getting the hat...');
+    await provider.send('tenderly_setStorageAt', [
+        CHIEF_ADDRESS,
+        ethers.utils.hexZeroPad(ethers.utils.hexValue(CHIEF_HAT_SLOT), 32),
+        ethers.utils.hexZeroPad(SPELL_ADDRESS, 32),
+    ]);
```

Comment:
Could we try to impersonate the pause proxy instead of relying on `setStorageAt`?
Seems doable at first sight: https://docs.tenderly.co/simulations-and-forks/reference/tenderly-fork-customization-via-api/how-to-impersonate-any-address-as-a-simulation-sender

However I did not dig too much.

Reply:
Actually, never mind... impersonating won't solve the issue here. `lift` is permissionless, it requires sufficient approvals only.

We'd need to keep in mind that there's a new Chief implementation coming and the storage layout might change there, so this script would have to be updated.

Reply:
To my understanding, because pause proxy have to do a delegate call, we can't impersonate pause proxy, we have to impersonate pause to call [pause-proxy's `exec`](https://github.com/dapphub/ds-pause/blob/0763eafcf926fd2e073aee5f047f3decb842231c/src/pause.sol#L114-L121). Then, [as `eta` is technically part of the spell state](https://github.com/makerdao/dss-exec-lib/blob/69b658f35d8618272cd139dfc18c5713caf6b96b/src/DssExec.sol#L41), it anyway have to be scheduled first. Maybe I'm also missing something, but impersonating pause proxy was also my first approach, which didn't worked out.

The other, classic approach would be to not use `setStorageAt` for the cheif's hat, but to mint enough mkr and then follow regular process of casting. Despite minting also requires `setStorageAt` under the hood, tenderly actually have more reliable abstraction of [`tenderly_setErc20Balance`](https://docs.tenderly.co/devnets/advanced/custom-rpc-methods#tenderly_seterc20balance). I actually would prefer this way, since then we can also use the same code to cast spell outside of the tenderly fork, but it will require more code 🙂

Reply:
Yeah, you are totally right. I'm gonna resolve this comment.
We just need to keep in mind that an update to the Chief might imply in maintenance for this script.